.github/workflows/build.yml

```diff
@@ -163,7 +163,7 @@ jobs:
           buildkitd-config-inline: |
             [registry."ghcr.io"]
       - name: Login to DockerHub
-        if: ${{ steps.should.outputs.RUN == 'true' }}
+        if: ${{ steps.should.outputs.RUN == 'true' && secrets.DOCKERHUB_USERNAME != '' && secrets.DOCKERHUB_TOKEN != '' }}
```

Comment:
I added this, as it is just for rate limited reasons, but forks don't need to have access to the docker hub creds.